### DISCUSSION TITLE
making auto generated gcodes for tool changing optional

### DISCUSSION THREAD

**Discussion by PandaPilot** [supermerill/SuperSlicer#2536] [Community contribution]
This gcode is generated every time after changing tools
<
G10 P2 S230 ; set temperature
M116 ; wait for temperature to be reached
>
Is it possible to make this optional so it does not interfere with the E3D Toolchanger's tool changing gcodes? (Note: I think this problem is unique to multi heater machines as M116 with no modifiers/parameters forces the printer to wait for all the standby heaters to cool.)

**Comment by supermerill** [comment]
Complicated. The change has to work for everything and everyone.
